app/queries/participant_rating_ranks_query.rb

```
@@ -6,16 +6,25 @@ def initialize(relation = Participant.all)
   end
 
   def call()
-    ranks = @relation.where('rating is not null').select('id, rank() over (order by rating desc) as current_ranking, ranking as previous_ranking, ranking_change').reorder('ranking asc')
+    @relation.where('fixed_rating is not null').map do |participant|
+      user_final_rating = UserRating.where("participant_id=#{participant.id} and rating is not null").reorder('created_at desc').first
+      time_difference = (Time.current.to_date - user_final_rating[:created_at].to_date)
+      time_difference = time_difference.to_i
+      total_number_of_days = 365
+      updated_rating = participant.fixed_rating * (Math.exp(-time_difference/total_number_of_days))
+      participant.update({rating: updated_rating})
```

You should use `update!` method to not fail silently.